src/lib/server/session.cpp

```diff
@@ -107,7 +108,10 @@ void Session::_handle_simple_query() {
   const auto execution_information = QueryHandler::execute_pipeline(query, _send_execution_info);
 
   if (!execution_information.error.empty()) {
-    _postgres_protocol_handler->send_error_message(execution_information.error);
+    const auto error_message = ErrorMessage{{PostgresMessageType::HumanReadableError, execution_information.error},
+                                            {PostgresMessageType::SqlstateCodeError,
+                                             std::to_string(static_cast<uint32_t>(ErrorCode::TransactionRolledBack))}};
+    _postgres_protocol_handler->send_error_message(error_message);
```

Comment:
How do we know that this is a rollback?

Reply:
It looks as if PostgreSQL does not enforce any particular execution_info format - it is just a plain string message. So, how about splitting the error field into two fields 'error code' and 'error message'. They can be set separately in the query handler, where we want the condition to have an additional 'otherwise/unknown' case. Here we check for the error code and act accordingly.